test/e2e/arktos/multi_tenancy/cmd/testrunner.go

```diff
@@ -0,0 +1,172 @@
+	testConfig         framework.TestConfig
+	testSuiteFiles     []string
+	invalidTestSuites  []string
+	validTestSuites    []*framework.TestSuite
+	testSuiteDirFlag   string
+	testSuiteFileFlag  string
+	commonVariableFlag string
+)
+
+func initFlags() {
+	basedir, _ := filepath.Abs(filepath.Dir(os.Args[0]))
+	defaultTestSuiteDir := filepath.Join(basedir, "test_suites")
+
+	flag.IntVar(&testConfig.MaxRetryCount, "MaxRetryCount", 10, "maximal retry counts of a test command")
+	flag.IntVar(&testConfig.MaxRetryInterval, "MaxRetryInterval", 60, "maximal retry interval in seconds")
+	flag.IntVar(&testConfig.MaxTimeOut, "MaxTimeOut", 300, "maximal timeout in seconds allowed for a test command")
+	flag.IntVar(&testConfig.DefaultRetryCount, "DefaultRetryCount", 0, "default retry counts of a test command")
+	flag.IntVar(&testConfig.DefaultRetryInterval, "DefaultRetryInterval", 5, "default retry interval in seconds")
+	flag.IntVar(&testConfig.DefaultTimeOut, "DefaultTimeOut", 2, "default timeout in seconds allowed for a test command")
+	flag.BoolVar(&testConfig.Verbose, "Verbose", false, "Extra logging if true")
+
+	flag.StringVar(&testSuiteDirFlag, "TestSuiteDir", defaultTestSuiteDir, "The directory of test suite files")
+	flag.StringVar(&testSuiteFileFlag, "TestSuiteFiles", "", "The test suite files")
+	flag.StringVar(&commonVariableFlag, "CommonVar", "", "Common variable definition used across test suites.")
+
+	flag.Parse()
+
+	validateFlags()
+}
+
+func validateFlags() {
+	if errs := testConfig.Validate(); !errs.IsEmpty() {
+		framework.LogError("\nThe test config is invalid: %v \n", errs)
+		os.Exit(1)
+	}
+
+	if strings.TrimSpace(testSuiteFileFlag) == "" {
+		framework.LogError("\nNo test suite file is specified.  (%v) \n", testSuiteFileFlag)
+		os.Exit(1)
+	}
+
+	for _, file := range strings.Fields(testSuiteFileFlag) {
+		testSuiteFiles = append(testSuiteFiles, filepath.Join(testSuiteDirFlag, strings.TrimSpace(file)))
+	}
+
+	for _, tsf := range testSuiteFiles {
+		if framework.FileExists(tsf) == false {
+			framework.LogError("\nTest suite file %q is missig.\n", tsf)
+			os.Exit(1)
+		}
+	}
+
+	if len(commonVariableFlag) > 0 {
+		testConfig.CommonVariables = make(map[string]string)
+		for _, kv := range strings.Split(commonVariableFlag, ",") {
+			if len(strings.Split(kv, ":")) != 2 {
+				framework.LogError("\n(%q) is not a valid variable definition. Ignored\n", kv)
+				continue
+			}
+
+			parts := strings.Split(kv, ":")
+			if strings.TrimSpace(parts[0]) == "" {
+				framework.LogError("\n(%q) is not a valid variable definition. Ignored\n", kv)
+				continue
+			}
+
+			testConfig.CommonVariables[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+		}
+	}
+}
+
+func verifyLocalClusterUp() {
+	var test framework.TestCase
+	test.Command = "kubectl get nodes"
+
+	errList := test.Run(&testConfig)
+	if !errList.IsEmpty() {
+		framework.LogError("\nArktos cluster is not up for test. Or you have don't have cluster admin privilege.\n")
```

Comment:
have don't have -> have not had

Reply:
also print out errList here for user's debugging benefit

Reply:
#1: fixed.